lib/backend/backend_test.go

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) 2016 Tigera, Inc. All rights reserved.
+
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package backend_test
+
+import (
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"github.com/projectcalico/libcalico-go/lib/backend/model"
+
+	"github.com/projectcalico/libcalico-go/lib/api"
+	"github.com/projectcalico/libcalico-go/lib/backend/etcd"
+	"github.com/projectcalico/libcalico-go/lib/client"
+	"github.com/projectcalico/libcalico-go/lib/testutils"
+)
+
+// Setting BackendType to etcdv2.
+var etcdType api.BackendType = "etcdv2"
+
+// Setting localhost as the etcd endpoint location since that's where `make run-etcd` runs it.
+var etcdConfig = &etcd.EtcdConfig{
+	EtcdEndpoints: "http://127.0.0.1:2379",
+}
+
+var _ = Describe("Backend tests", func() {
+
+	FDescribe("etcd GET/List Revision values", func() {
+		testutils.CleanEtcd()
+		c, _ := testutils.NewClient("")
+		ic := testutils.NewIPAMClient(true, etcdType, etcdConfig)
```

Comment:
I'm not a big fan of these tests mixing the use of the frontend client + backend client.

These are really backend FVs.  Instead of creating an IPAMClient, let's just use the backend.etcd client to create objects, read them back, and ensure a version is present.

Each test can be:

- Create
- Get
- List
- Delete


You can easily grab a backend client like this:
```
cfg := api.ClientConfig{BackendType: "etcdv2", BackendConfig: etcdConfig}
client, _:= client.New(cfg)
backendClient := client.backend
```